### PR TITLE
Add API reference and example for USBCDC_ECM class

### PR DESCRIPTION
**Pull request by juhaylinen**

### CODE REVIEW

**Comment by juhaylinen** [comment]
Documentation for PR https://github.com/ARMmbed/mbed-os/pull/9443

File: docs/api/usb/USBCDC_ECM.md
Comment (AnotherButler) [inline docs/api/usb/USBCDC_ECM.md]: Query: What are the typical use cases for this class? What are the common scenarios in which you'd use it?

File: docs/api/usb/USBCDC_ECM.md
Comment (AnotherButler) [inline docs/api/usb/USBCDC_ECM.md]: Are there plans to make this an importable example? If there's no set plan for this, please add this example to this repository: https://github.com/ARMmbed/mbed-os-examples-docs_only

File: docs/api/usb/USBCDC_ECM.md
Comment (AnotherButler) [inline docs/api/usb/USBCDC_ECM.md:11]: I'll wait to approve this PR until the code merges, and I can check this link.

File: docs/api/usb/USBCDC_ECM.md
Comment (AnotherButler) [inline docs/api/usb/USBCDC_ECM.md:3]: I'll wait to approve this PR until the code merges, and I can check this link.

**Review by AnotherButler — COMMENTED** [review]
Thanks for the PR. Please review my changes and address my queries.

**Comment by AnotherButler** [comment]
Does anyone know why this is failing Travis?

**Comment by cmonr** [comment]
@AnotherButler `apt-get update` is notoriously flaky.

Restarting the job in this instance is sufficient, but I would suggest having someone refactor `.travis.yml` at some point to remove the command instance.

**Comment by cmonr** [comment]
And naturally both @geky, @kegilbert, and I will be out until next month :grin:

File: docs/api/usb/USBCDC_ECM.md
Comment (juhaylinen) [reply docs/api/usb/USBCDC_ECM.md]: Typical applications are USB to Ethernet adapter and USB Web server that allows users to configure a USB device and to view data using a web browser.  The user needs to add required SW components on top of the USBCDC_ECM class like IP stack and Web server.

File: docs/api/usb/USBCDC_ECM.md
Comment (juhaylinen) [reply docs/api/usb/USBCDC_ECM.md]: Do you know what are the plans for other USB device examples? I can add the example if needed.

Comment (AnotherButler) [reply docs/api/usb/USBCDC_ECM.md]: We would like to make the other USB device examples importable examples instead of having them hardcoded in the documentation.

Comment (juhaylinen) [reply docs/api/usb/USBCDC_ECM.md]: PR created https://github.com/ARMmbed/mbed-os-examples-docs_only/pull/50

**Comment by kegilbert** [comment]
https://github.com/ARMmbed/mbed-os-5-docs/blob/development/.travis.yml#L22 this apt-get update? That was added for for the code snippet compilation validation. Looks like it has a safe-guard somewhat in place, but @geky would know more.

**Comment by 0xc0170** [comment]
@c1728p9 Can you please review?

**Review by 0xc0170 — APPROVED** [review]

**Comment by 0xc0170** [comment]
> Documentation for PR ARMmbed/mbed-os#9443

About to get integrated, for 5.13. Lets go ahead with this one to have it docs

**Comment by AnotherButler** [comment]
If I merge this before the Doxygen pages exist, it'll break my build. I'll merge as soon as the Doxygen's ready.

**Comment by mikter** [comment]
@AnotherButler and @c1728p9  are we ok to merge the documentation and example of the USB class pull requests. 
https://github.com/ARMmbed/mbed-os-examples-docs_only/pull/50

**Comment by AnotherButler** [comment]
I believe @0xc0170 requested changes on ARMmbed/mbed-os-examples-docs_only#50.
I can't merge this documentation until the Doxygen appears. If I merge with an embedded link to a Doxygen page that doesn't exist, it breaks my build, and no Mbed OS docs will be able to build.

**Comment by AnotherButler** [comment]
The Doxygen has merged 🎉 We only need the example changes, and then this is good to go

**Review by AnotherButler — APPROVED** [review]
Thanks so much for all of your hard work on this PR 👍 It looks great.

**Pull request merged**
